ui/src/criteria/concept.tsx

```diff
@@ -1,9 +1,33 @@
 import Typography from "@mui/material/Typography";
 import { DataGrid, GridColDef } from "@mui/x-data-grid";
 import React, { useContext, useEffect, useMemo, useState } from "react";
-import { useParams } from "react-router-dom";
-import { EntityInstancesApiContext } from "./apiContext";
-import { DataSet } from "./dataSet";
+import { EntityInstancesApiContext } from "../apiContext";
+import { Criteria, DataSet, Group } from "../dataSet";
+
+export class ConceptCriteria extends Criteria {
+  constructor(name: string, filter: string) {
+    super();
+    this.name = name;
+    this.filter = filter;
+  }
+
+  edit(dataSet: DataSet, group: Group): JSX.Element {
+    return (
```

Comment:
Hmm, I thought this was a model class but it also contains rendering logic - should it? Minimally I think it would be good to name the method to be more obvious related to React / rendering, e.g. `renderEdit`